File: 백준/Silver/1074. Z/Z.py
def row(r):
    if r == 0:      return 0
    elif r == 1:    return 2
    e = 0
    while r >= 2**e:
        e += 1
    e -= 1
    return row(r - 2**e) + 2**(e*2+1)

def col(c):
    if c == 0:      return 0
    elif c == 1:    return 1
    e = 0
    while c >= 2**e:
        e += 1
    e -= 1
    return col(c - 2**e) + 2**(e*2)
    
n,r,c = map(int,input().split())
print(row(r)+col(c))

File: 백준/Silver/1074. Z/Z.cc
#include <iostream>

using namespace std;

int n,r,c;

int Row(int row)
{
    if(row == 0)    return 0;
    if(row == 1)    return 2;
    int e = 0;
    while(row >= 1<<e)
    {
        e++;
    }
    e--;
    return (1<<(e*2+1)) + Row(row - (1<<e));    
}

int Col(int col)
{
    if(col == 0)    return 0;
    if(col == 1)    return 1;
    int e = 0;
    while(col >= 1<<e)
    {
        e++;
    }
    e--;
    return (1<<(e*2)) + Col(col - (1<<e));
}

int main()
{
    cin>>n>>r>>c;
    cout<<Row(r)+Col(c)<<endl;
}